File: Leetcode/53-Maximum-Subarray/MaximumSubarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    if (!nums || nums.length == 0) {
        return 0
    }
    var res = Number.MIN_SAFE_INTEGER
    var sum = 0
    for (i=0; i<nums.length; i++) {
        sum += nums[i]
        res = Math.max(res, sum)
        sum = Math.max(sum, 0)
    }
    return res
};

File: Leetcode/53-Maximum-Subarray/MaximumSubarray.go
import "math"

func maxSubArray(nums []int) int {
	if nums == nil || len(nums) == 0 {
		return 0
	}

	max := math.MinInt64
	sum := 0
	for _, val := range nums {
		sum += val
		max = int(math.Max(float64(max), float64(sum)))
		sum = int(math.Max(float64(sum), 0.0))
	}
	return max
}
